homeassistant/components/influx.py

```diff
@@ -0,0 +1,143 @@
+"""
+homeassistant.components.influx
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+InfluxDB component which allows you to send data to an Influx database.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/influx/
+
+Configuration:
+
+influx:
+  host: localhost
+  port: 8086
+  dbname: home_assistant
+  dbuser: DB_USER
+  dbuser_password: DB_USER_PASSWORD
+"""
+import logging
+import requests
+import socket
+
+import homeassistant.util as util
+from homeassistant.helpers import validate_config
+from homeassistant.const import (MATCH_ALL)
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = "influx"
+DEPENDENCIES = ['recorder']
+
+INFLUX_CLIENT = None
+
+DEFAULT_HOST = 'localhost'
+DEFAULT_PORT = 8086
+DEFAULT_DATABASE = 'home_assistant'
+
+REQUIREMENTS = ['influxdb==2.10.0']
+
+CONF_HOST = 'host'
+CONF_PORT = 'port'
+CONF_DB_NAME = 'database'
+CONF_USERNAME = 'username'
+CONF_PASSWORD = 'password'
+
+
+def setup(hass, config):
+    """ Setup the Influx component. """
+
+    from influxdb import exceptions
+
+    if not validate_config(config, {DOMAIN: ['host']}, _LOGGER):
+        return False
+
+    conf = config[DOMAIN]
+
+    host = conf[CONF_HOST]
+    port = util.convert(conf.get(CONF_PORT), int, DEFAULT_PORT)
+    dbname = util.convert(conf.get(CONF_DB_NAME), str, DEFAULT_DATABASE)
+    username = util.convert(conf.get(CONF_USERNAME), str)
+    password = util.convert(conf.get(CONF_PASSWORD), str)
+
+    global INFLUX_CLIENT
+
+    try:
+        INFLUX_CLIENT = Influx(host, port, username, password, dbname)
+    except (socket.gaierror, requests.exceptions.ConnectionError):
+        _LOGGER.error("Database is not accessible. "
+                      "Please check your entries in the configuration file.")
+        return False
+
+    try:
+        INFLUX_CLIENT.create_database(dbname)
+    except exceptions.InfluxDBClientError:
+        _LOGGER.info("Database '%s' already exists", dbname)
+
+    INFLUX_CLIENT.switch_user(username, password)
+    INFLUX_CLIENT.switch_database(dbname)
+
+    def event_listener(event):
+        """ Listen for new messages on the bus and sends them to Influx. """
+        event_data = event.as_dict()
+        json_body = []
+
+        if event_data['event_type'] is not 'time_changed':
```

Comment:
So it looks like you only care about `state_changed` events. Please filter out all others event types, then there will be no need for the `try…except` block. See the logbook code for [an example how to handle events](https://github.com/balloob/home-assistant/blob/dev/homeassistant/components/logbook.py#L165-L175).

Should look something like this:

```python
# For top of the file
from homeassistant.core import State
from homeassistant.const import EVENT_STATE_CHANGED


if event_data['event_type'] != EVENT_STATE_CHANGED:
    return

state = State.from_dict(event.data.get('new_state'))
```